### DISCUSSION TITLE
How to change the users/me to make it set a cookie

### DISCUSSION THREAD

**Discussion by diegocgaona** [fastapi-users/fastapi-users#1397] [Q&A] [answered]
I would like to use the /user/me route to renew my cookie, if the user is logged in and call the /user/me it will setcookie with a new max-age.
How can I do it?

Thanks in advance!!

**Comment by hgalytoby** [comment]
I am not familiar with using cookies. I spent some time researching how to use cookies with fastapi-users, and this should meet your expectations.

The old cookie values can still be used as long as they haven't expired.

```py
class MyCookieTransport(CookieTransport):
    def update_cookie(self, response: Response, token: str):
        response.set_cookie(
            self.cookie_name,
            token,
            max_age=self.cookie_max_age,
            path=self.cookie_path,
            domain=self.cookie_domain,
            secure=self.cookie_secure,
            httponly=self.cookie_httponly,
            samesite=self.cookie_samesite,
        )


async def update_cookie(
    request: Request,
    response: Response,
    user: User = Depends(current_active_user),
):
    if request.url.path.endswith('/me') and request.method == "GET":
        strategy = auth_backend.get_strategy()
        token = await strategy.write_token(user=user)
        auth_backend.transport.update_cookie(
            response=response,
            token=token,
        )


app.include_router(
    fastapi_users.get_users_router(UserRead, UserUpdate),
    prefix="/users",
    tags=["users"],
    dependencies=[Depends(update_cookie)]
)
```

**Comment by diegocgaona** [comment]
Thanks @hgalytoby , but I'm having the error:

```python
access_token = await self.database.create(access_token_dict)
                         ^^^^^^^^^^^^^^^^^^^^
AttributeError: 'Depends' object has no attribute 'create'
```
The error came from:
`token = await strategy.write_token(user)`

I'm using Beanie, I think I would need to change something to use it (I don't know why it tries to create with Depends),

**Comment by hgalytoby** [comment, marked as answer]
@diegocgaona 

I didn't realize you were using `DatabaseStrategy`.

This is the first time I know about `DatabaseStrategy`, I just tried it and I have no problem.

```py
async def update_cookie(
    request: Request,
    response: Response,
    strategy: DatabaseStrategy = Depends(auth_backend.get_strategy),
    user: User = Depends(current_active_user),
):
    if request.url.path.endswith('/me') and request.method == "GET":
        token = await strategy.write_token(user=user)
        auth_backend.transport.update_cookie(
            response=response,
            token=token,
        )
```

You may refer to this

https://github.com/fastapi-users/fastapi-users/blob/master/fastapi_users/router/auth.py#L55

https://github.com/fastapi-users/fastapi-users/blob/master/fastapi_users/authentication/backend.py#L42

The `Depends` question, I don't think it's a difficult one, you should spend some time on it.

**Comment by diegocgaona** [comment]
Thanks @hgalytoby 😊 , you last answer helped me and it is worked fine.
I have some difficulties with security, but now I have what I need
